gguf-py/scripts/gguf-template.py

```diff
@@ -0,0 +1,140 @@
+"""
+gguf-template.py - example file to extract the chat template from the models metadata
+"""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import sys
+from pathlib import Path
+
+import jinja2
+
+# Necessary to load the local gguf package
+if "NO_LOCAL_GGUF" not in os.environ and (Path(__file__).parent.parent.parent / 'gguf-py').exists():
+    sys.path.insert(0, str(Path(__file__).parent.parent))
+
+from gguf.constants import Keys
+from gguf.gguf_reader import GGUFReader  # noqa: E402
+
+# Configure logging
+logger = logging.getLogger("gguf-chat-template")
+
+
+def get_chat_template(model_file: str) -> str:
+    reader = GGUFReader(model_file)
+
+    # Available keys
+    logger.debug("Detected model metadata!")
+    logger.debug("Outputting available model fields:")
+    for key in reader.fields.keys():
+        logger.debug(key)
+
+    # Access the 'chat_template' field directly using its key
+    chat_template_field = reader.fields.get(Keys.Tokenizer.CHAT_TEMPLATE)
```

Comment:
Mind adding support for multiple chat templates, like in Command R (tool_use and rag)?

Will require some dummy `tools` and `documents` arguments to the template as well, so perhaps some way to arbitrarily add those?

Reply:
I can look into it. Shy on time, but I'll see what I can do.

Reply:
If you don't have the time, don't worry about it, maybe add it as a TODO? Just Would Have Been Nice To Have™

Reply:
TODO then. I took a mini vacation this past week, so I had a bit of extra time. Back to work though. Can add it in another future PR.